src/main/java/io/vertx/openapi/contract/OpenAPIContractBuilder.java

```diff
@@ -0,0 +1,234 @@
+   *
+   * @param contractFiles A map that contains all additional contract files.
+   * @return The builder, for a fluent interface.
+   */
+  public OpenAPIContractBuilder setAdditionalContractFiles(Map<String, String> contractFiles) {
+    additionalContractFiles.clear();
+    for (var e : contractFiles.entrySet()) {
+      putAdditionalContractFile(e.getKey(), e.getValue());
+      additionalContracts.remove(e.getKey());
+    }
+    return this;
+  }
+
+
+  /**
+   * Adds a contract that is referenced by the main contract. This method can be
+   * called multiple times to add multiple referenced contracts.
+   *
+   * @param key     The unique key for the contract.
+   * @param content The parsed contract.
+   * @return The builder, for a fluent interface
+   */
+  public OpenAPIContractBuilder putAdditionalContract(String key, JsonObject content) {
+    additionalContracts.put(key, content);
+    additionalContractFiles.remove(key);
+    return this;
+  }
+
+  /**
+   * Uses the contracts from the provided map to resolve referenced contracts.
+   * Replaces all previously put contracts by {@link #putAdditionalContract(String, JsonObject)}.
+   * If the same key is used also replaces the contracts set by {@link #putAdditionalContractFile(String, String)}
+   * and {@link #setAdditionalContractFiles(Map)}.
+   *
+   * @param contracts A map that contains all additional contract files.
+   * @return The builder, for a fluent interface.
+   */
+  public OpenAPIContractBuilder setAdditionalContracts(Map<String, JsonObject> contracts) {
+    additionalContracts.clear();
+    for (var e : contracts.entrySet()) {
+      putAdditionalContract(e.getKey(), e.getValue());
+      additionalContractFiles.remove(e.getKey());
+    }
+    return this;
+  }
+
+  /**
+   * Builds the contract.
+   *
+   * @return The contract.
+   */
+  public Future<OpenAPIContract> build() {
+    if (contractFile == null && contract == null) {
+      return Future.failedFuture(new OpenAPIContractBuilderException("Neither a contract file or a contract is set. One of them must be set."));
+    }
+
+    Future<JsonObject> readContract = contractFile == null
+      ? Future.succeededFuture(contract)
+      : Utils.readYamlOrJson(vertx, contractFile);
+
+    var resolvedContracts = Future
+      .succeededFuture(additionalContracts)
+      .compose(x -> readContractFiles()
+        .map(r -> {
+          var all = new HashMap<>(x);
+          all.putAll(r);
+          return all;
+        }));
+
+    return Future.all(readContract, resolvedContracts)
+      .compose(x -> {
+        JsonObject contract = x.resultAt(0);
+        Map<String, JsonObject> other = x.resultAt(1);
+        return buildOpenAPIContract(contract, other);
+      });
+  }
+
+  private Future<OpenAPIContract> buildOpenAPIContract(JsonObject unresolvedContract,
+                                                       Map<String, JsonObject> additionalContractFiles) {
```

Comment:
```suggestion
                                                       Map<String, JsonObject> additionalContracts) {
```

This are no longer contract files, they are resolved